certbot/plugins/script.py

```diff
@@ -0,0 +1,172 @@
+"""Standalone Authenticator."""
+import logging
+import os
+import sys
+
+import zope.interface
+
+from subprocess import Popen, PIPE
+
+from acme import challenges
+
+from certbot import errors
+from certbot import interfaces
+
+from certbot.plugins import common
+
+logger = logging.getLogger(__name__)
+
+
+CHALLENGES = ["http-01", "dns-01"]
+
+
+@zope.interface.implementer(interfaces.IAuthenticator)
+@zope.interface.provider(interfaces.IPluginFactory)
+class Authenticator(common.Plugin):
+    """Script authenticator
+
+    calls user defined script to perform authentication and
+    optionally cleanup.
+
+    """
+
+    description = "Authenticate using user provided script(s)"
```

Comment:
Unfortunately more_info() can't be used, as PluginEntryPoint can't access class methods, only class variables. I implemented long_description variable, and PluginEntryPoint property with the same name, that uses long_description if available, falling back to description if not.  Also put it to use in CLI.